controllers/ovnkubeconfig_controller.go

```
@@ -234,3 +253,76 @@ func (r *OVNKubeConfigReconciler) getLocalOvnkubeImage() (string, error) {
 	}
 	return ds.Spec.Template.Spec.Containers[0].Image, nil
 }
+
+func (r *OVNKubeConfigReconciler) syncMachineConfigObjs(mcpName string) error {
+	var err error
+	foundMc := &mcfgv1.MachineConfig{}
+	foundMcp := &mcfgv1.MachineConfigPool{}
+	mcp := &mcfgv1.MachineConfigPool{}
+	mcp.Name = mcpName
+	mcSelector, err := metav1.ParseToLabelSelector(fmt.Sprintf("%s in (worker,%s)", mcfgv1.MachineConfigRoleLabelKey, dpuMcRole))
+	nodeSelector := metav1.SetAsLabelSelector(labels.Set{dpuNodeLabel: ""})
+	mcp.Spec = mcfgv1.MachineConfigPoolSpec{
+		MachineConfigSelector: mcSelector,
+		NodeSelector:          nodeSelector,
+	}
+	if mcpName == "master" || mcpName == "worker" {
+		return fmt.Errorf("%s pools is not allowed", mcpName)
```

This function will always get called for the master and worker machineconfigpools though, won't it? So you shouldn't call this an error; just `return nil` to silently ignore them.

---

No, this function shall only be invoked for the custom MCP name user specified in the OVNKubeConfig CR. Users shall never use a built-it MCP here.